crates/bevy_animation/src/lib.rs

```diff
@@ -691,217 +677,271 @@ where
-    if !verify_no_ancestor_player(maybe_parent, parents) {
-        warn!("Animation player on {:?} has a conflicting animation player on an ancestor. Cannot safely animate.", root);
-        return;
+impl Plugin for AnimationPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_asset::<AnimationClip>()
+            .register_asset_reflect::<AnimationClip>()
+            .register_type::<AnimationPlayer>()
+            .register_type::<AnimationTarget>()
+            .add_systems(
+                PostUpdate,
+                (
+                    advance_animations,
+                    animate_targets,
+                    expire_animation_transitions,
+                )
+                    .chain()
+                    .before(TransformSystem::TransformPropagate),
+            );
     }
+}
 
-    let mut any_path_found = false;
-    for (path, bone_id) in &animation_clip.paths {
-        let cached_path = &mut animation.path_cache[*bone_id];
-        let curves = animation_clip.get_curves(*bone_id).unwrap();
-        let Some(target) = entity_from_path(root, path, children, names, cached_path) else {
-            continue;
+impl PlayingAnimation {
+    fn apply(
+        &self,
+        clips: &Assets<AnimationClip>,
+        weight: f32,
+        target_context: &mut AnimationTargetContext,
+    ) {
+        let Some(clip) = clips.get(&self.animation_clip) else {
+            warn!(
+                "Couldn't find the animation clip for {:?} ({:?})",
+                target_context.id, target_context.name,
+            );
+            return;
         };
-        any_path_found = true;
-        // SAFETY: The verify_no_ancestor_player check above ensures that two animation players cannot alias
-        // any of their descendant Transforms.
-        //
-        // The system scheduler prevents any other system from mutating Transforms at the same time,
-        // so the only way this fetch can alias is if two AnimationPlayers are targeting the same bone.
-        // This can only happen if there are two or more AnimationPlayers are ancestors to the same
-        // entities. By verifying that there is no other AnimationPlayer in the ancestors of a
-        // running AnimationPlayer before animating any entity, this fetch cannot alias.
-        //
-        // This means only the AnimationPlayers closest to the root of the hierarchy will be able
-        // to run their animation. Any players in the children or descendants will log a warning
-        // and do nothing.
-        let Ok(mut transform) = (unsafe { transforms.get_unchecked(target) }) else {
-            continue;
+
+        let Some(curves) = clip.get_curves(target_context.target.id) else {
+            return;
         };
-        // SAFETY: As above, there can't be other AnimationPlayers with this target so this fetch can't alias
-        let mut morphs = unsafe { morphs.get_unchecked(target) }.ok();
+
+        self.apply_curves(curves, weight, target_context);
```

Comment:
Is this used anywhere else? Can we just inline the code and save a function call (mostly for readability)?